### ISSUE TITLE
Missing release notes for v3.25.1

### ISSUE THREAD

**Issue by rittneje** [newrelic/go-agent#791]
There are no real release notes for v3.25.1. It just says "Corrects an error in the release process for 3.25.0." There is no release entry at all for v3.25.0.

In addition, there is no entry for v3.25.0 or v3.25.1 on https://docs.newrelic.com/docs/release-notes/agent-release-notes/go-release-notes/

I did finally find them in the [CHANGELOG](https://github.com/newrelic/go-agent/blob/master/CHANGELOG.md) but it is strange they are missing from the usual locations.

**Comment by nr-swilloughby** [comment]
We'll fix this in the next release. Essentially we started the process to release 3.25.0 but it didn't go correctly and the wrong commit got tagged, so essentially 3.25.0 doesn't exist since it wasn't completed successfully and contained effectively no changes. However, the tag was visible publicly long enough we couldn't just move the tags so instead we successfully released the code but bumped the version to 3.25.1 so it has a distinct version number and git tag. At the same time, we updated `go.mod` to retract 3.25.0 so the Go ecosystem won't try to use it.

The release notes could have been more clear about that. This has been fixed for the upcoming release, with the actual changes for 3.25.1 moved to the 3.25.1 section, and the 3.25.0 section simply containing a note explaining the above.
